src/hotspot/os/bsd/os_perf_bsd.cpp

```diff
@@ -246,15 +265,15 @@ int CPUPerformanceInterface::CPUPerformance::cpu_loads_process(double* pjvmUserL
   long jvm_system_nanos = 0;
 
   for (size_t i = 0; i < length / sizeof(*lproc); i ++) {
-     struct rusage procusg = lproc[i].ki_rusage;
-     jvm_user_nanos += procusg.ru_utime.tv_usec + (1000000 * procusg.ru_utime.tv_sec);
-     jvm_system_nanos += procusg.ru_stime.tv_usec + (1000000 * procusg.ru_stime.tv_sec);
+     jvm_user_nanos += lproc[i].KI_UTIME_USEC + (1000000 * lproc[i].KI_UTIME_SEC);
```

Comment:
FWIW, I believe this is why I'm seeing orders (3, specifically) of magnitude differences between the user and system (kernel) values and the overall load.  `jvm_user_nanos` and `jvm_system_nanos` are clearly in microseconds, not nanoseconds.  However, they are divided by the difference in nanoseconds (see the implementation of `now_in_nanos`).  Both of these values need multiplication by 1000 (alternatively, one could convert to getting the current time in microseconds (e.g. `now_in_micros`), since all `now_in_nanos` is doing is calculating the current time in microseconds and multiplying it by 1000).

Reply:
Yes, good catch on the conversion bug.